cmd/savepcdfiles/main.go

```diff
@@ -69,10 +82,23 @@ func mainWithArgs(ctx context.Context, args []string, logger golog.Logger) error
 		Attributes: config.AttributeMap{"device_path": usbDevices[0].Path},
 	}
 
-	return savePCDFiles(ctx, int(argsParsed.Port), lidarDevice, logger)
+	// Create new data directory
+	newpath := filepath.Join(".", "data")
+
+	err := os.RemoveAll(newpath)
+	if err != nil {
+		return errors.New("error deleting data directory")
+	}
+
+	err = os.MkdirAll(newpath, 0777)
```

Comment:
So the issue is that, with the current setup for SLAM we are using those saved files. The current process is to generate the saved files, once we have a complete run that we want to test on then we will copy those files over to a different directory that cartographer is looking at. 

I'm fine moving to the temp directory but the deleting of files will have to be on hold until we finish testing SLAM and we start building the full pipeline from RPLiDAR -> SLAM. Additionally, I see the /tmp/ directory is not empty at present

![image](https://user-images.githubusercontent.com/92594844/153507067-5da3ade5-8f26-4545-bd4f-5c430f20a3d6.png)

Are we able to clear those files everytime we startup a new run on rplidar? I dont want pcd files to get confused with one another

Reply:
I think there might be some confusion, based on the fact that Jeremy chose to delete the directory with data every time the program restarts.

The program is currently aimed at collecting data; and we want to keep that since we need to use it in the next step, which is running slam, like Jeremy already said.

I am currently working on a second soon-to-be-opened PR, which will not delete the folder of data. The reason being: I don't want to accidentally delete my hard-earned data just because I restart the rplidar.

Instead, in my upcoming PR, I am allowing the user to define the name of the folder where they want to save their data (the default being `data`), and if the directory already exists, then the new data will be added to the saved data.

I don't mind saving the data in a temp directory - although since it is actually data we want to keep - and keep track of - wouldn't we want to have it in the same directory as the code that's running the rplidar?